lib/facter/resolvers/linux/docker_uptime.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module Facter
+  module Resolvers
+    module Linux
+      class DockerUptime < BaseResolver
+        @semaphore = Mutex.new
+        @fact_list ||= {}
+        class << self
+          private
+
+          def post_resolve(fact_name)
+            @fact_list.fetch(fact_name) { detect_uptime(fact_name) }
+          end
+
+          def detect_uptime(fact_name)
+            days, hours, minutes, seconds = extract_uptime_from_docker
+            total_seconds = convert_to_seconds(days, hours, minutes, seconds)
+            @fact_list = Utils::UptimeHelper.create_uptime_hash(total_seconds)
+
+            @fact_list[fact_name]
+          end
+
+          def extract_uptime_from_docker
+            # time format [dd-][hh:]mm:ss
+            time = Facter::Core::Execution.execute('ps -o etime= -p "1"', logger: log)
+            extracted_time = time.split(/[-:]/)
+
+            reversed_time = extracted_time.reverse
+            seconds = reversed_time[0]&.to_i
+            minutes = reversed_time[1]&.to_i || 0
+            hours = reversed_time[2]&.to_i || 0
+            days = reversed_time[3]&.to_i || 0
```

Comment:
do you need the & check and the || 0? if reversed_time is empty array => reversed_time[int] will be empty and will result in 0 if .to_i method is called

Reply:
```
2.6.3 :004 > arra =[ 1 ]
 => [1]
2.6.3 :005 > arra[2]
 => nil
2.6.3 :006 > arra[2]&.to_i
 => nil
```

Reply:
```>> nil.to_i   #=> 0```

Reply:
Changed